### PR TITLE
✨BackButton ➕ Testing

### LINKED ISSUE #20: Back Button
#### Description

the button to go back from the Map page to pages

#### Screenshots

image

#### Tasks

Include specific tasks in the order they need to be done in. Include links to specific lines of code where the task should happen at.

- [x] create the Back button 
- [x] link to pages

### PR DESCRIPTION
**Pull request by JGieseWeb**
Create the Back button and function.
+ implement testing function

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,28 +10,37 @@
...
     "prepare": "husky install",
-    "test": "tsc && prettier --check . && eslint . && stylelint \"**/*.css\"",
+    "test": "tsc && prettier --check . && eslint . && stylelint \"**/*.css\" && npm run jest",
     "build": "npm run server:build && npm run client:build && npm run storybook:build",
-    "start": "node dist/server.js"
+    "start": "node dist/server.js",
+    "jest": "jest"
   },
   "devDependencies": {
     "@babel/core": "^7.14.3",
```
Comment (lajanzen) [inline package.json]: Maybe you want to deinstall babel core? We used `npm remove @babel/core`

**Review by lajanzen — APPROVED** [review]
🚀

File: src/app/components/BackButton/BackButton.test.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  expect(screen.getByRole("button").innerHTML).toBe(
+    '<svg xmlns="http://www.w3.org/2000/svg" width="24.743" height="23.34" viewBox="0 0 24.743 23.34"><path d="M28.223,15.75H13.177l5.836-5.583a2.326,2.326,0,0,0,0-3.178,2.079,2.079,0,0,0-3.038,0L6.258,16.411a2.142,2.142,0,0,0-.633,1.575v.028a2.142,2.142,0,0,0,.633,1.575l9.71,9.422a2.079,2.079,0,0,0,3.037,0,2.326,2.326,0,0,0,0-3.178L13.17,20.25H28.216A2.2,2.2,0,0,0,30.368,18,2.178,2.178,0,0,0,28.223,15.75Z" transform="translate(-5.625 -6.33)" fill="var(--navbar-icon-color)"></path></svg>'
+  );
+});
+
+test("calls onClick", () => {
+  const handleClick = jest.fn();
+  render(<BackButton onClick={handleClick}></BackButton>);
+
+  fireEvent.click(screen.getByRole("button"));
```
Comment (Morten-Moeller) [inline src/app/components/BackButton/BackButton.test.tsx]: The fireEvent is kind of old logic for jest. You should use [userEvent](https://testing-library.com/docs/ecosystem-user-event/) (You have to import it then)
```suggestion
  userEvent.click(screen.getByRole("button"));
```

File: src/app/components/BackButton/BackButton.test.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import { fireEvent, render, screen } from "@testing-library/react";
```
Comment (Morten-Moeller) [inline src/app/components/BackButton/BackButton.test.tsx]:
```suggestion
import { userEvent, render, screen } from "@testing-library/react";
```

**Review by Morten-Moeller — COMMENTED** [review]
Good job, just a suggestion for the jest part.

File: package.json (hunk 1 above)
Comment (JGieseWeb) [reply package.json]: Thanks, I did it 👍

File: src/app/components/BackButton/BackButton.test.tsx (hunk 2 above)
Comment (JGieseWeb) [reply src/app/components/BackButton/BackButton.test.tsx]: Thanks, I like to be up-to-date 👍

**Pull request merged**